zipline/pipeline/loaders/events.py

```
@@ -36,196 +42,135 @@
-                    if not isinstance(v.index, pd.DatetimeIndex):
-                        raise ValueError(
-                            SERIES_NO_DTINDEX_ERROR.format(sid=k)
-                        )
-                    self.events_by_sid[k] = pd.DataFrame({
-                        list(self.expected_cols)[0]: v})
-                elif isinstance(v, pd.DatetimeIndex):
-                    if not infer_timestamps:
-                        raise ValueError(
-                            DTINDEX_NOT_INFER_TS_ERROR.format(sid=k)
-                        )
-                    self.events_by_sid[k] = pd.DataFrame({
-                        list(self.expected_cols)[0]: v
-                    }, index=[dates[0]] * len(v))
-                else:
-                    # We expect 1 column, but we got something other than a
-                    # Series, DatetimeIndex, or DataFrame.
-                    raise ValueError(
-                        WRONG_SINGLE_COL_DATA_FORMAT_ERROR.format(sid=k)
-                    )
-            else:
-                # We expected multiple columns, but we got something other
-                # than a DataFrame.
-                raise ValueError(
-                    WRONG_MANY_COL_DATA_FORMAT_ERROR.format(sid=k)
-                )
-        self.events_by_sid = {sid: df.dropna(subset=[self.event_date_col]) for
-                              sid, df in self.events_by_sid.items()}
-        self.dataset = dataset
-
-    def get_loader(self, column):
-        if column in self.dataset.columns:
-            return getattr(self, "%s_loader" % column.name)
-        raise ValueError("Don't know how to load column '%s'." % column)
-
-    def load_adjusted_array(self, columns, dates, assets, mask):
-        return merge(
-            self.get_loader(column).load_adjusted_array(
-                [column], dates, assets, mask
+                 events,
+                 next_value_columns,
+                 previous_value_columns):
```

We'll need a check at load time no matter what to give a useful error if the engine passes us columns we don't know how to load, but we could do a pre-emptive check on our events in the constructor to fail faster.